ckan/lib/plugins.py

```diff
@@ -99,24 +105,22 @@ def register_package_plugins(map):
                 raise ValueError("More than one fallback "
                                  "IDatasetForm has been registered")
             _default_package_plugin = plugin
-
         for package_type in plugin.package_types():
             # Create a connection between the newly named type and the
             # package controller
-
-            map.connect('%s_search' % package_type, '/%s' % package_type,
-                        controller='package', action='search')
-
-            map.connect('%s_new' % package_type, '/%s/new' % package_type,
-                        controller='package', action='new')
-            map.connect('%s_read' % package_type, '/%s/{id}' % package_type,
-                        controller='package', action='read')
-
-            for action in ['edit', 'authz', 'history']:
-                map.connect('%s_%s' % (package_type, action),
-                            '/%s/%s/{id}' % (package_type, action),
-                            controller='package',
-                            action=action)
+            blueprint = Blueprint(
+                package_type,
+                dataset.dataset.import_name,
+                url_prefix='/{}'.format(package_type),
+                url_defaults={'package_type': package_type})
+            blueprint.add_url_rule(
+                '/', view_func=dataset.search, strict_slashes=False)
+            blueprint.add_url_rule(
+                '/new', view_func=dataset.CreateView.as_view(str(u'new')))
+            blueprint.add_url_rule('/<id>', view_func=dataset.read)
+            blueprint.add_url_rule(
+                '/edit/<id>', view_func=dataset.EditView.as_view(str(u'edit')))
```

Comment:
We are missing some routes here, follow and activity. Plugins should have the same routes as the main blueprints

Reply:
The resource URLS are not being customized with the dataset type either. To be clear this is the current behaviour in master, but I wonder is now is the time to properly support all dataset related URLs? cc @wardi